Refer to target_link_libraries instead of ament_target_dependencies

sloretz opened the pull request:
I think this is complete, but I haven't self-reviewed it yet. Leaving as a draft for now.

sloretz committed:
Make docs refer to target_link_libraries instead of ament_target_dependencies

Signed-off-by: Shane Loretz <sloretz@intrinsic.ai>

christophebedard commented on source/Tutorials/Intermediate/Writing-an-Action-Server-Client/Cpp.rst:
```diff
@@ -221,11 +221,11 @@ Open up ``custom_action_cpp/CMakeLists.txt``, and add the following right after
+    ${custom_action_interfaces_TARGETS}
+    rclcpp::rclcpp
+    rclcpp_action::rclcpp_action
+    rclcpp_components::components)
```
I think this might be `rclcpp_components::component`, without the `s` at the end.

```suggestion
    rclcpp_components::component)
```

Same for `action_client` below.

sloretz replied:
Fixed in 052618b8 and d39b4280

fujitatomoya reviewed (approved):
lgtm with @christophebedard comments.

sloretz committed:
components -> component

https://github.com/ros2/rclcpp/blob/f78ed952b27acc63ef8022d78cb816c309a9ca3d/rclcpp_components/CMakeLists.txt#L25

Co-authored-by: Christophe Bedard <bedard.christophe@gmail.com>
Signed-off-by: Shane Loretz <shane.loretz@gmail.com>

sloretz committed:
components -> component

https://github.com/ros2/rclcpp/blob/f78ed952b27acc63ef8022d78cb816c309a9ca3d/rclcpp_components/CMakeLists.txt#L25

Signed-off-by: Shane Loretz <sloretz@intrinsic.ai>

christophebedard commented:
Do you also plan to mention the deprecation in the Kilted release notes like this section in the Jazzy release notes? https://docs.ros.org/en/rolling/Releases/Release-Jazzy-Jalisco.html#changes-since-the-iron-release

sloretz commented:
> Do you also plan to mention the deprecation in the Kilted release notes like this section in the Jazzy release notes? https://docs.ros.org/en/rolling/Releases/Release-Jazzy-Jalisco.html#changes-since-the-iron-release

Ah thank you for the reminder! https://github.com/ros2/ros2_documentation/pull/5232

ros-discourse commented:
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/best-practices-for-replacing-ament-target-dependencies-in-kilted-while-maintaining-compatibility/43938/5
